### ISSUE TITLE
Pluggable rule engine

### ISSUE THREAD

**Issue by bloomonkey** [autotraderuk/oliver-twist#3]
It would be good if adding rules was as simple as telling CLI to look in a directory containing `.py` scripts that look something like:

```python
@Rule(link="http://example.com/my-rule")
def obey_my_rule():
   ...
   
```

**Comment by dorzey** [comment]
Can we keep the scope of this to be discovering rules within this repo?

**Comment by dorzey** [comment]
@bloomonkey given we've merged #7; can we close this?
